{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "0c8025a3-6cfe-4866-919c-5bbb905ae5a3",
   "metadata": {},
   "outputs": [],
   "source": [
    "#######This document is for Pooled samples report for the EPI team#######\n",
    "\n",
    "######First I create a dictionary of reportable SNPs so that I could later use it to filter out unreportable SNP samples####\n",
    "\n",
    "dictv={}\n",
    "#####The voinew3.csv file contains a table of reportable SNPs########\n",
    "with open (\"voinew3.csv\", \"r\") as v1:\n",
    "    ####I loop through lines in the voi file ###\n",
    "    for lines in v1:\n",
    "        ###I would like to skip the header####\n",
    "        if \"Ref\" not in lines:\n",
    "            ######I add Gene name and SNP####\n",
    "            dictv[lines.split(\",\")[2]+lines.split(\",\")[3]+lines.split(\",\")[4].strip(\"\\n\")]=\"exist\"\n",
    "\n",
    "###Open raw output table file \"Annotations.csv\" from Geneious#####\n",
    "import itertools\n",
    "###### Assign letters to the amino acid names #########\n",
    "#####This is a dictionary assigning three letters to amino acid letter #####\n",
    "AAdic = {\n",
    "\"Ala\": \"A\",\n",
    "\"Arg\": \"R\",\n",
    "\"Asn\": \"N\",\n",
    "\"Asp\": \"D\",\n",
    "\"Asx\": \"B\",\n",
    "\"Cys\": \"C\",\n",
    "\"Glu\": \"E\",\n",
    "\"Gln\": \"Q\",\n",
    "\"Glx\": \"Z\",\n",
    "\"Gly\": \"G\",\n",
    "\"His\": \"H\",\n",
    "\"Ile\": \"I\",\n",
    "\"Leu\": \"L\",\n",
    "\"Lys\": \"K\",\n",
    "\"Met\": \"M\",\n",
    "\"Phe\": \"F\",\n",
    "\"Pro\": \"P\",\n",
    "\"Ser\": \"S\",\n",
    "\"Thr\": \"T\",\n",
    "\"Trp\": \"W\",\n",
    "\"Tyr\": \"Y\",\n",
    "\"Val\": \"V\",\n",
    "\"STOP\": \"X\",\n",
    "}\n",
    "#######Convert combination of bases to a codon############\n",
    "######This is coverting three base combination into a coddon####\n",
    "#####It would create dicionary of three bases conversion to codons######\n",
    "currentcodon = \"\"\n",
    "keywords = [\"\".join(i) for i in itertools.product([\"T\", \"G\", \"A\", \"C\"], repeat=3)]\n",
    "codondic = {}\n",
    "for x in keywords:\n",
    "    currentcodon = \"\"\n",
    "    for y in x:\n",
    "        currentcodon += y\n",
    "    # print(currentcodon)\n",
    "    if currentcodon == \"TTT\" or currentcodon == \"TTC\":\n",
    "        # print(\"True\")\n",
    "        codondic[currentcodon] = \"Phe\"\n",
    "    if currentcodon == \"TTA\" or currentcodon == \"TTG\":\n",
    "        codondic[currentcodon] = \"Leu\"\n",
    "    if (\n",
    "        currentcodon == \"CTT\"\n",
    "        or currentcodon == \"CTC\"\n",
    "        or currentcodon == \"CTA\"\n",
    "        or currentcodon == \"CTG\"\n",
    "    ):\n",
    "        codondic[currentcodon] = \"Leu\"\n",
    "    if currentcodon == \"ATT\" or currentcodon == \"ATC\" or currentcodon == \"ATA\":\n",
    "        codondic[currentcodon] = \"Ile\"\n",
    "    if currentcodon == \"ATG\":\n",
    "        codondic[currentcodon] = \"Met\"\n",
    "    if (\n",
    "        currentcodon == \"GTT\"\n",
    "        or currentcodon == \"GTC\"\n",
    "        or currentcodon == \"GTA\"\n",
    "        or currentcodon == \"GTG\"\n",
    "    ):\n",
    "        codondic[currentcodon] = \"Val\"\n",
    "    if (\n",
    "        currentcodon == \"TCT\"\n",
    "        or currentcodon == \"TCC\"\n",
    "        or currentcodon == \"TCA\"\n",
    "        or currentcodon == \"TCG\"\n",
    "    ):\n",
    "        codondic[currentcodon] = \"Ser\"\n",
    "    if (\n",
    "        currentcodon == \"CCT\"\n",
    "        or currentcodon == \"CCC\"\n",
    "        or currentcodon == \"CCA\"\n",
    "        or currentcodon == \"CCG\"\n",
    "    ):\n",
    "        codondic[currentcodon] = \"Pro\"\n",
    "    if (\n",
    "        currentcodon == \"ACT\"\n",
    "        or currentcodon == \"ACC\"\n",
    "        or currentcodon == \"ACA\"\n",
    "        or currentcodon == \"ACG\"\n",
    "    ):\n",
    "        codondic[currentcodon] = \"Thr\"\n",
    "    if (\n",
    "        currentcodon == \"GCT\"\n",
    "        or currentcodon == \"GCC\"\n",
    "        or currentcodon == \"GCA\"\n",
    "        or currentcodon == \"GCG\"\n",
    "    ):\n",
    "        codondic[currentcodon] = \"Ala\"\n",
    "    if currentcodon == \"TAT\" or currentcodon == \"TAC\":\n",
    "        codondic[currentcodon] = \"Tyr\"\n",
    "    if currentcodon == \"CAT\" or currentcodon == \"CAC\":\n",
    "        codondic[currentcodon] = \"His\"\n",
    "    if currentcodon == \"CAA\" or currentcodon == \"CAG\":\n",
    "        codondic[currentcodon] = \"Gln\"\n",
    "    if currentcodon == \"AAT\" or currentcodon == \"AAC\":\n",
    "        codondic[currentcodon] = \"Asn\"\n",
    "    if currentcodon == \"AAA\" or currentcodon == \"AAG\":\n",
    "        codondic[currentcodon] = \"Lys\"\n",
    "    if currentcodon == \"GAT\" or currentcodon == \"GAC\":\n",
    "        codondic[currentcodon] = \"Asp\"\n",
    "    if currentcodon == \"GAA\" or currentcodon == \"GAG\":\n",
    "        codondic[currentcodon] = \"Glu\"\n",
    "    if currentcodon == \"TGT\" or currentcodon == \"TGC\":\n",
    "        # print(\"I am correct\")\n",
    "        codondic[currentcodon] = \"Cys\"\n",
    "    if currentcodon == \"TGG\":\n",
    "        codondic[currentcodon] = \"Trp\"\n",
    "    if (\n",
    "        currentcodon == \"AGA\"\n",
    "        or currentcodon == \"AGG\"\n",
    "        or currentcodon == \"CGT\"\n",
    "        or currentcodon == \"CGC\"\n",
    "        or currentcodon == \"CGA\"\n",
    "        or currentcodon == \"CGG\"\n",
    "    ):\n",
    "        codondic[currentcodon] = \"Arg\"\n",
    "    if currentcodon == \"AGT\" or currentcodon == \"AGC\":\n",
    "        codondic[currentcodon] = \"Ser\"\n",
    "    if (\n",
    "        currentcodon == \"GGT\"\n",
    "        or currentcodon == \"GGC\"\n",
    "        or currentcodon == \"GGA\"\n",
    "        or currentcodon == \"GGG\"\n",
    "    ):\n",
    "        codondic[currentcodon] = \"Gly\"\n",
    "    if currentcodon == \"TGA\" or currentcodon == \"TAA\" or currentcodon == \"TAG\":\n",
    "        codondic[currentcodon] = \"STOP\"\n",
    "        \n",
    "count220=0\n",
    "dict3={}\n",
    "#####Open Geneious raw output file which is Annotation.csv####\n",
    "####This is the ulfiltered file######\n",
    "with open (\"Annotations.csv\", \"r\") as r1:\n",
    "    ####We will write filterd information on GuineaAnalysis_Pooled_EP######\n",
    "    with open(\"GuineaAnalysis_pooled_EP.csv\" , \"w\") as w1:\n",
    "        ####We will write the header first with given information#####\n",
    "        w1.write(\"Sample,Pooled,Year,SITE,TreatmentDay,GENE,G_annotation,COVERAGE,VAF,VF,SNP,TYPE\\n\")\n",
    "        count=0\n",
    "        ####We will create a dictionary with area code in it so that####\n",
    "        ####We can infer name of region from the sample ID#####\n",
    "        dict1={}\n",
    "        dict1[\"Ha\"]=\"Hamdalaye\"\n",
    "        dict1[\"Do\"]=\"Dorota\"\n",
    "        dict1[\"Ma\"]=\"Maferinyah\"\n",
    "        dict1[\"La\"]=\"Lay-Sarè\"\n",
    "        dict1[\"LS\"]=\"Lay-Sarè\"\n",
    "        dict2={}\n",
    "        #####We will loop through each line in the raw Geneious output file######\n",
    "        for lines in r1:\n",
    "            count+=1\n",
    "            ######Check if there is Polymorphism in the line from the table#######\n",
    "            ######If there is polymorphism then that means there is variant######\n",
    "            ######According to Geneious as geneious report variants as polymorphisms####\n",
    "            ######We will loop through Geneious raw output for variants first####\n",
    "            if \"Polymorphism\" in lines and \"Variants\" in lines:\n",
    "                ####sample name is first column right before the _ string#####\n",
    "                ####We will filer out unnecessary information from sample name#####\n",
    "                samplename=lines.split(\",\")[0].split(\"_\")[0]\n",
    "                ####Sample gene is the second coulmmn in the line####\n",
    "                ###We will assign genes to the given variants of interest####\n",
    "                samplegene=lines.split(\",\")[1]\n",
    "                ###We want to correct name for DHPS####\n",
    "                if \"DHPS\" in samplegene:\n",
    "                    samplegene=\"DHPS\"\n",
    "                ###We want to correct name of MT CYTB####\n",
    "                if \"mitochondrial\" in samplegene:\n",
    "                    samplegene=\"CYTB\"\n",
    "                ###Sample AA change is column 17 which has the amino acid changes####\n",
    "                sampleAAchange=lines.split(\",\")[16]\n",
    "                #####For silent mutation when there is no amino acid change####\n",
    "                #####We check if there is mutation change or not####\n",
    "                ####If there is no amino acid change we considered it silent mutation####\n",
    "                if sampleAAchange==\"\" and lines.split(\",\")[26]!=\"\":\n",
    "                    ####For silent mutation we have to use three base changes instead of aminoacid letter change###\n",
    "                    ####Due to how Geneious reports it#####\n",
    "                    ####We use the previously created codon dictionary and amino acid dictionary to get the amino acid letter for the three bases changes####\n",
    "                    a=AAdic[codondic[lines.split(\",\")[26][0:3]]]\n",
    "                    b=AAdic[codondic[lines.split(\",\")[26][-3::]]]\n",
    "                    ####C is the location of amino acid change####\n",
    "                    c=lines.split(\",\")[20]\n",
    "                    AAchangereport=(a+c+b)\n",
    "                    ####Again we get the sample variant allele frequency and variant frequency depending on the location in the geneious raw output table####\n",
    "                    sampleVAF=(lines.split(\"M05039\")[0]).split(\",\")[-4]\n",
    "                    sampleVF=(lines.split(\"M05039\")[0]).split(\",\")[-2]\n",
    "                    ####We also get the average coverage depednign on which column it is###\n",
    "                    ####I need to further investigate using average coverage vs maximum coverage####\n",
    "                    sampleCoverage=lines.split(\",\")[28]\n",
    "                        ####Use a dictionary to prevent overlaps######\n",
    "                    if samplename[9]==\"p\" and (samplename,AAchangereport[0:-1]) not in dict3 and AAchangereport in dictv:\n",
    "                        ####We want to assign to dictionary3 for preventing duplicates in the future####\n",
    "                        dict3[samplename,AAchangereport[0:-1]]=\"exist\"\n",
    "                        #####If the 1A is in that postion of sample ID we assign treatment day 1#####\n",
    "                        #####We assign information based on the header we wrote previously#####\n",
    "                        if samplename[6:8]==\"1A\":\n",
    "                            w1.write(samplename+\",\"+\"Pooled\"+\",\"+samplename[0:2]+\",\"+dict1[samplename[4:6]]+\",\"+\"01\"+\",\"+samplegene+\",\"+AAchangereport+\",\"+sampleCoverage+\",\"+sampleVAF+\",\"+sampleVF+\",\"+AAchangereport+\",\"+\"silent\"+\"\\n\")\n",
    "                        ####Otherwise we assign the treatment day based on name####\n",
    "                        else:\n",
    "                            w1.write(samplename+\",\"+\"Pooled\"+\",\"+samplename[0:2]+\",\"+dict1[samplename[4:6]]+\",\"+samplename[6:8]+\",\"+samplegene+\",\"+AAchangereport+\",\"+sampleCoverage+\",\"+sampleVAF+\",\"+sampleVF+\",\"+AAchangereport+\",\"+\"silent\"+\"\\n\")\n",
    "                ####If there is Amino acid change reported combine it with the c variable which is codon position####\n",
    "                ####to create a SNP change report #####\n",
    "                if sampleAAchange!=\"\":\n",
    "                    ####Based on spacing you could figurout aminoacid letters for before changing after changing#####\n",
    "                    a=sampleAAchange.split(\" \")[0]\n",
    "                    b=sampleAAchange.split(\" \")[2]\n",
    "                    ####C is the aminoacid position where change has happened######\n",
    "                    c=lines.split(\",\")[20]\n",
    "                    AAchangereport=(a+c+b)\n",
    "                    ###Next We will assign variant allele frequency to a column####\n",
    "                    sampleVAF=(lines.split(\"M05039\")[0]).split(\",\")[-4]\n",
    "                    ###Next We will assign variant frequency to the next column####\n",
    "                    sampleVF=(lines.split(\"M05039\")[0]).split(\",\")[-2]\n",
    "                    ####We want to assign coverage information to the next column####\n",
    "                    sampleCoverage=lines.split(\",\")[28]\n",
    "                    ###We want to make sure these are Guinea sample which has GN in its name####\n",
    "                    if \"GN\" in samplename:\n",
    "                        ####Use a dictionary to prevent overlaps######\n",
    "                        ####We want to use condition that there shouldn't be letter p in that position#####\n",
    "                        ####To identify individual samples also we don't want to process duplicates such as ####\n",
    "                        ####duplicate wildtypes or variants once we processed them because there can be only either variant or wildtype####\n",
    "                        ####Also for EPI report the SNP has to be in the previously created dictionary#####\n",
    "                        ####We want to make the dictionary without the last letter which is changed amino acid because ####\n",
    "                        ####We want to prevent situation such as same amino acid change and same wildtype but different mutation change #####\n",
    "                        if samplename[9]==\"p\" and (samplename,AAchangereport[0:-1]) not in dict3 and AAchangereport in dictv:\n",
    "                            ####We want to assign to dictionary3 for preventing duplicates in the future####\n",
    "                            dict3[samplename,AAchangereport[0:-1]]=\"exist\"\n",
    "                            #####If the 1A is in that postion of sample ID we assign treatment day 1#####\n",
    "                            #####We assign information based on the header we wrote previously#####\n",
    "                            w1.write(samplename+\",\"+\"Pooled\"+\",\"+samplename[0:2]+\",\"+dict1[samplename[4:6]]+\",\"+samplename[6:8]+\",\"+samplegene+\",\"+AAchangereport+\",\"+sampleCoverage+\",\"+sampleVAF+\",\"+sampleVF+\",\"+AAchangereport[1::]+\",\"+\"mutation\"+\"\\n\")\n",
    "\n",
    "####We open the annotation file again for identifying pure wildtpye this time####                            \n",
    "with open (\"Annotations.csv\", \"r\") as r1:\n",
    "    with open(\"GuineaAnalysis_pooled_EP.csv\" , \"a\") as w1:\n",
    "        ####We go through each line in the annotation file####\n",
    "        for lines in r1:\n",
    "            ####This time conditional would be only for coverage####\n",
    "            ####If there is coverage and it is not 0 that means####\n",
    "            ####It is wildtype#####\n",
    "            if \"Coverage:\" in lines:\n",
    "                ####sample name is first column right before the _ string#####\n",
    "                ####We will filer out unnecessary information from sample name#####\n",
    "                samplename=lines.split(\",\")[0].split(\"_\")[0]\n",
    "                ####Sample gene is the second coulmmn in the line####\n",
    "                ###We will assign genes to the given wildtypes of interest####\n",
    "                samplegene=lines.split(\",\")[1]\n",
    "                ###We want to correct name for DHPS####\n",
    "                if \"DHPS\" in samplegene:\n",
    "                    samplegene=\"DHPS\"\n",
    "                ###We want to correct name of MT CYTB####\n",
    "                if \"mitochondrial\" in samplegene:\n",
    "                    samplegene=\"CYTB\"\n",
    "                ####For lines containing Coverage instead of polymorphism###\n",
    "                ###The aminoacid change is reported at different column####\n",
    "                ###It is reported at the last column as trackerSNP####\n",
    "                AAchangereport=lines[0:-12].split(\",\")[-1]\n",
    "                ###Since this is wildtype sampleVAF would be 0%####\n",
    "                ###Sample variant frequency is also 0#####\n",
    "                sampleVAF=\"0%\"\n",
    "                sampleVF=\"0\"\n",
    "                ####For lines containing Coverage instead of polymorphism###\n",
    "                ###The sample coverage is reported at different column####\n",
    "                sampleCoverage=lines.split(\",\")[17]\n",
    "                ####Again we are only interested in Guinea samples####\n",
    "                if \"GN\" in samplename:\n",
    "                    ####Since this is individual report for EPI team we are gonna find individuals that are not p####\n",
    "                    ####at a given position of the naming schema###\n",
    "                    if samplename[9]==\"p\" and (samplename,AAchangereport[0:-1]) not in dict3 and AAchangereport in dictv:\n",
    "                        ###We want to setup a dictionary so that we wouldn't create duplicate report from the variant report made previously#####\n",
    "                        dict3[samplename,AAchangereport[0:-1]]=\"exist\"\n",
    "                        ####we just assign the given treatment day in the naming schema#####\n",
    "                        w1.write(samplename+\",\"+\"Pooled\"+\",\"+samplename[0:2]+\",\"+dict1[samplename[4:6]]+\",\"+samplename[6:8]+\",\"+samplegene+\",\"+AAchangereport+\",\"+sampleCoverage+\",\"+sampleVAF+\",\"+sampleVF+\",\"+AAchangereport[0:-1]+\",\"+\"wildtype\"+\"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0bd2004d-949c-4fde-bf91-776d6622a9a4",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1e227511-60e0-4e5e-acec-bec5c3a09b1e",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
