{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "5a54c818-1256-4073-b55b-13aea3b74497",
   "metadata": {},
   "outputs": [],
   "source": [
    "#########Extract individual information about variant frequency of individuals###########\n",
    "#########Extract pooled information about variant frequency of pooled samples###########\n",
    "######First open the previous file which contains both individual and pooled samples info with just pooled size added####\n",
    "with open (\"GuineaAnalysis_ps.csv\", \"r\") as r2:\n",
    "    count=0\n",
    "    dict1={}\n",
    "    for lines in r2:\n",
    "        count+=1\n",
    "        if count>1:\n",
    "            ####Set up a dictionary to add the weights and sample sizes to calcualte WVAF later#######\n",
    "            #####This for loop is for the mutation with notification of -> in it######\n",
    "            if lines.split(\",\")[2]+\",\"+lines.split(\",\")[4]+\",\"+lines.split(\",\")[3].strip(\"\\n\").strip(\" \") in dict1:\n",
    "                if lines.split(\",\")[-3]!=\"0%\":\n",
    "                    if \"->\" in lines.split(\",\")[-3]:\n",
    "                        #print(lines)\n",
    "                        ####This is for mutation that is silent####\n",
    "                        ####Assign dictionary with given combined pooled information#####\n",
    "                        ####If there is arrow inthe column assign it as mutation###\n",
    "                        ####Mutation is considered as VAF* pooled size to be added in the second part of list####\n",
    "                        ####Also add pool size to the first part of list#####\n",
    "                        ####wild type is consiered as adding 1*pooled size for first part of the list####\n",
    "                        ####Wildtype only adds 0 to the VAF sum which is the second part of the list####\n",
    "                        ####Only add VAF and poolsize to the samples when the item is already in the dictionary######\n",
    "                        dict1[lines.split(\",\")[2]+\",\"+lines.split(\",\")[4]+\",\"+lines.split(\",\")[3].strip(\"\\n\").strip(\" \")]=[x + y for x, y in zip(dict1[lines.split(\",\")[2]+\",\"+lines.split(\",\")[4]+\",\"+lines.split(\",\")[3].strip(\"\\n\").strip(\" \")],[1*float(lines.split(\",\")[-1]),1*float(lines.split(\",\")[-3][-4::].strip(\"%\"))*float(lines.split(\",\")[-1])])]\n",
    "                    else:\n",
    "                        ####This is for mutation that is not silent#####\n",
    "                        ####If arrow not in given column we consider it as wildtype#####\n",
    "                        ####wild type is consiered as adding 1*pooled size for first part of the list####\n",
    "                        ####Wildtype only adds 0 to the VAF sum which is the second part of the list####\n",
    "                        ####Only add VAF and poolsize to the samples when the item is already in the dictionary######\n",
    "                        dict1[lines.split(\",\")[2]+\",\"+lines.split(\",\")[4]+\",\"+lines.split(\",\")[3].strip(\"\\n\").strip(\" \")]=[x + y for x, y in zip(dict1[lines.split(\",\")[2]+\",\"+lines.split(\",\")[4]+\",\"+lines.split(\",\")[3].strip(\"\\n\").strip(\" \")],[1*float(lines.split(\",\")[-1]),1*float(lines.split(\",\")[-3].strip(\"%\"))*float(lines.split(\",\")[-1])])]\n",
    "                    \n",
    "            ###If the item is not in dictionary just assign the VAF and pooled size#### \n",
    "            ###Consiser mutation that VAF is not 0 and it is not in dictionary###%####\n",
    "            if lines.split(\",\")[2]+\",\"+lines.split(\",\")[4]+\",\"+lines.split(\",\")[3].strip(\"\\n\").strip(\" \") not in dict1:\n",
    "                if lines.split(\",\")[-3]!=\"0%\":\n",
    "                    dict1[lines.split(\",\")[2]+\",\"+lines.split(\",\")[4]+\",\"+lines.split(\",\")[3].strip(\"\\n\").strip(\" \")]=[1*float(lines.split(\",\")[-1]),1*float(lines.split(\",\")[-3].strip(\"%\"))*float(lines.split(\",\")[-1])]\n",
    "            ###Consiser wildtype that VAF is 0 and it is in dictionary already###%####\n",
    "            if lines.split(\",\")[2]+\",\"+lines.split(\",\")[4][0:-1]+\",\"+lines.split(\",\")[3].strip(\"\\n\").strip(\" \") in dict1:\n",
    "                if lines.split(\",\")[-3]==\"0%\":\n",
    "                    dict1[lines.split(\",\")[2]+\",\"+lines.split(\",\")[4][0:-1]+\",\"+lines.split(\",\")[3].strip(\"\\n\").strip(\" \")]=[x + y for x, y in zip(dict1[lines.split(\",\")[2]+\",\"+lines.split(\",\")[4][0:-1]+\",\"+lines.split(\",\")[3].strip(\"\\n\").strip(\" \")],[1*float(lines.split(\",\")[-1]),0])]\n",
    "            ###Consiser wildtype that VAF is 0 and it is in dictionary already###%####\n",
    "            if lines.split(\",\")[2]+\",\"+lines.split(\",\")[4][0:-1]+\",\"+lines.split(\",\")[3].strip(\"\\n\").strip(\" \") not in dict1:\n",
    "                if lines.split(\",\")[-3]==\"0%\":\n",
    "                    dict1[lines.split(\",\")[2]+\",\"+lines.split(\",\")[4][0:-1]+\",\"+lines.split(\",\")[3].strip(\"\\n\").strip(\" \")]=[1*float(lines.split(\",\")[-1]),0]\n",
    "                    \n",
    "\n",
    "#########Extract individual information about variant frequency of individuals###########\n",
    "#########Extract pooled information about variant frequency of pooled samples###########\n",
    "######First open the previous file which contains both individual and pooled samples info with just pooled size added####\n",
    "######We want to create duplicate dictionary to be used later#####\n",
    "#####The way we create dictionary is same as previous for loop#####\n",
    "#####This loop is for mutation without -> sign in the notification#####\n",
    "with open (\"GuineaAnalysis_ps.csv\", \"r\") as r2:\n",
    "    count=0\n",
    "    dict3={}\n",
    "    for lines in r2:\n",
    "        count+=1\n",
    "        if count>1:\n",
    "            if lines.split(\",\")[2]+\",\"+lines.split(\",\")[4]+\",\"+lines.split(\",\")[3].strip(\"\\n\").strip(\" \") in dict3:\n",
    "                if lines.split(\",\")[-3]!=\"0%\":\n",
    "                    dict3[lines.split(\",\")[2]+\",\"+lines.split(\",\")[4]+\",\"+lines.split(\",\")[3].strip(\"\\n\").strip(\" \")]=[x + y for x, y in zip(dict3[lines.split(\",\")[2]+\",\"+lines.split(\",\")[4]+\",\"+lines.split(\",\")[3].strip(\"\\n\").strip(\" \")],[1*int(lines.split(\",\")[-1]),0])]\n",
    "            if lines.split(\",\")[2]+\",\"+lines.split(\",\")[4]+\",\"+lines.split(\",\")[3].strip(\"\\n\").strip(\" \") not in dict3:\n",
    "                if lines.split(\",\")[-3]!=\"0%\":\n",
    "                    dict3[lines.split(\",\")[2]+\",\"+lines.split(\",\")[4]+\",\"+lines.split(\",\")[3].strip(\"\\n\").strip(\" \")]=[1*int(lines.split(\",\")[-1]),0]\n",
    "            if lines.split(\",\")[2]+\",\"+lines.split(\",\")[4][0:-1]+\",\"+lines.split(\",\")[3].strip(\"\\n\").strip(\" \") in dict3:\n",
    "                if lines.split(\",\")[-3]==\"0%\":\n",
    "                    dict3[lines.split(\",\")[2]+\",\"+lines.split(\",\")[4][0:-1]+\",\"+lines.split(\",\")[3].strip(\"\\n\").strip(\" \")]=[x + y for x, y in zip(dict3[lines.split(\",\")[2]+\",\"+lines.split(\",\")[4][0:-1]+\",\"+lines.split(\",\")[3].strip(\"\\n\").strip(\" \")],[0,1*int(lines.split(\",\")[-1])])]\n",
    "            if lines.split(\",\")[2]+\",\"+lines.split(\",\")[4][0:-1]+\",\"+lines.split(\",\")[3].strip(\"\\n\").strip(\" \") not in dict3:\n",
    "                if lines.split(\",\")[-3]==\"0%\":\n",
    "                    dict3[lines.split(\",\")[2]+\",\"+lines.split(\",\")[4][0:-1]+\",\"+lines.split(\",\")[3].strip(\"\\n\").strip(\" \")]=[0,1*int(lines.split(\",\")[-1])]\n",
    "                                \n",
    "                        \n",
    "####Once we create dictionaries we want to combine them####\n",
    "####We combine the lists created for wildtype and mutations####\n",
    "####We wanted to use the combination information to calculate WVAF####\n",
    "with open (\"weighted_geneious.csv\", \"w\") as w1:\n",
    "    w1.write(\"Site,G_annotation,Gene,Number of Samples,SNP,WVAF,Number of Mutants, Number of wildtypes\\n\")\n",
    "    dict2={}\n",
    "    dict4={}\n",
    "    for item in dict1:\n",
    "        if item.split(\",\")[1][-1].isdigit()==False:\n",
    "            dict2[item]=dict3[item]\n",
    "            dict4[item]=dict1[item]\n",
    "    for item2 in dict2:\n",
    "        for item in dict3:\n",
    "            if item.split(\",\")[1][-1].isdigit():\n",
    "                if item.split(\",\")[0]==item2.split(\",\")[0] and item.split(\",\")[2]==item2.split(\",\")[2] and item.split(\",\")[1] in item2.split(\",\")[1]:\n",
    "                    ####If there is silent mutation###\n",
    "                    #####C469C is special case in which other same mutations happens####\n",
    "                    #####It adds up wildtypes for same amino acid positioned mutations####\n",
    "                    #####Thus, we divide up the total wildtypes to the half because of duplicate in addition######\n",
    "                    if \"C469\" in item:\n",
    "                        dict2[item2]=[x + y for x, y in zip(dict2[item2],[dict3[item][0],dict3[item][1]/2])]\n",
    "                        dict4[item2]=[x + y for x, y in zip(dict4[item2],[(dict3[item][0]+dict3[item][1]/2),dict1[item][1]])]\n",
    "                    else:\n",
    "                        dict2[item2]=[x + y for x, y in zip(dict2[item2],dict3[item])]\n",
    "                        dict4[item2]=[x + y for x, y in zip(dict4[item2],dict1[item])]\n",
    "    ####This part is actually writing the mutations and lines to the new document file####\n",
    "    ####After adding up all the wildtypes and mutations to a dictioanry while we write it\n",
    "    ####We wan to divide the total added up VAF * pooled size / total sample size####\n",
    "    for item in dict2:\n",
    "        if dict2[item][0]>0:\n",
    "            if item.split(\",\")[1][0]==item.split(\",\")[1][-1]:\n",
    "                w1.write(item+\",\"+str(dict2[item][1]+dict2[item][0])+\",\"+item.split(\",\")[1]+\",\"+str(round((dict4[item][1]/(dict4[item][0])),2))+\"%\"+\",\"+str(dict2[item][0])+\",\"+str(dict2[item][1])+\"\\n\")\n",
    "            else:\n",
    "                w1.write(item+\",\"+str(dict2[item][1]+dict2[item][0])+\",\"+item.split(\",\")[1][1::]+\",\"+str(round((dict4[item][1]/(dict4[item][0])),2))+\"%\"+\",\"+str(dict2[item][0])+\",\"+str(dict2[item][1])+\"\\n\")\n",
    "        if dict2[item][0]==0:\n",
    "            w1.write(item+\",\"+str(dict2[item][1]+dict2[item][0])+\",\"+item.split(\",\")[1][0:-1]+\",\"+str(round((dict2[item][0]/(dict2[item][1]+dict2[item][0]))*100,2))+\"%\"+\",\"+str(dict2[item][0])+\",\"+str(dict2[item][1])+\"\\n\")\n",
    "\n",
    "####We loop again to process wildtpyes####\n",
    "####This loop is for wildtypes meaning no variant at all#####\n",
    "with open (\"GuineaAnalysis_ps.csv\", \"r\") as r2:\n",
    "    count=0\n",
    "    dict1={}\n",
    "    ###This is where we create another dictionary just for the wildtypes####\n",
    "    for lines in r2:\n",
    "        count+=1\n",
    "        if count>1:\n",
    "            if lines.split(\",\")[2]+\",\"+lines.split(\",\")[4]+\",\"+lines.split(\",\")[3].strip(\"\\n\").strip(\" \") in dict1:\n",
    "                    if lines.split(\",\")[-3]==\"0%\":\n",
    "                        dict1[lines.split(\",\")[2]+\",\"+lines.split(\",\")[4]+\",\"+lines.split(\",\")[3].strip(\"\\n\").strip(\" \")]=[x + y for x, y in zip(dict1[lines.split(\",\")[2]+\",\"+lines.split(\",\")[4]+\",\"+lines.split(\",\")[3].strip(\"\\n\").strip(\" \")],[0,1*int(lines.split(\",\")[-1])])]\n",
    "                    if lines.split(\",\")[-3]!=\"0%\":\n",
    "                        dict1[lines.split(\",\")[2]+\",\"+lines.split(\",\")[4]+\",\"+lines.split(\",\")[3].strip(\"\\n\").strip(\" \")]=[x + y for x, y in zip(dict1[lines.split(\",\")[2]+\",\"+lines.split(\",\")[4]+\",\"+lines.split(\",\")[3].strip(\"\\n\").strip(\" \")],[1*int(lines.split(\",\")[-1]),0])]\n",
    "            if lines.split(\",\")[2]+\",\"+lines.split(\",\")[4]+\",\"+lines.split(\",\")[3].strip(\"\\n\").strip(\" \") not in dict1:\n",
    "                    if lines.split(\",\")[-3]==\"0%\":\n",
    "                        dict1[lines.split(\",\")[2]+\",\"+lines.split(\",\")[4]+\",\"+lines.split(\",\")[3].strip(\"\\n\").strip(\" \")]=[0,1*int(lines.split(\",\")[-1])]\n",
    "                    if lines.split(\",\")[-3]!=\"0%\":\n",
    "                        dict1[lines.split(\",\")[2]+\",\"+lines.split(\",\")[4]+\",\"+lines.split(\",\")[3].strip(\"\\n\").strip(\" \")]=[1*int(lines.split(\",\")[-1]),0]\n",
    "\n",
    "#####Again we write all the wildtypes in this for loop\n",
    "with open (\"weighted_geneious.csv\", \"a\") as w1:\n",
    "    for item in dict1:\n",
    "        if dict1[item][0]==0:\n",
    "            w1.write(item+\",\"+str(dict1[item][1]+dict1[item][0])+\",\"+item.split(\",\")[1][0:-1]+\",\"+str(round((dict1[item][0]/(dict1[item][1]+dict1[item][0]))*100,2))+\"%\"+\",\"+str(dict1[item][0])+\",\"+str(dict1[item][1])+\"\\n\")\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "93f144bf-d6a6-4c85-a2d2-99fe0948ce69",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "056578ec-b192-4250-81a5-329577d6a919",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
