{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "e292b979-e7de-47e5-b6e6-f3795c44400b",
   "metadata": {},
   "source": [
    "## Geneious analysis for individual samples from raw Geneious output, \"Annotation.csv\" \n",
    "\n",
    " ### Required packages\n",
    " - No specific package required \n",
    " \n",
    " ### Inputs\n",
    " - Geneious SNP analysis of _k13_, _crt_, _mdr1_, _dhfr_, _dhps_, and _cytb_\n",
    " - Documentation on Geneious analysis can be found: Readme.md\n",
    " - Geneious outputs were modified to GuineaAnalysis_Individual.csv from \"Annotation.csv\"\n",
    " \n",
    " \n",
    " ### Data structure \n",
    " - [Long-form](https://seaborn.pydata.org/tutorial/data_structure.html#long-form-vs-wide-form-data) \n",
    "     - Each variable is a column \n",
    "\n",
    "         - \"Sample\" = *AMD ID*, including associated meta-data for each sample\n",
    "             - AMD ID and bit code key is found under MS Teams > Domestic > Files > Sample Naming > Sample_naming_key.pptx  \n",
    "\n",
    "             - Key: **Year Country State/Site DayofTreatment Treatment SampleID Genus SampleType GeneMarker-8bitcode SampleSeqCount**\n",
    "\n",
    "                 - Example:\n",
    "                     - Individual sequenced sample ID: 17GNDo00F0001PfF1290 = 2017 Guinea Dorota Day0 AS+AQ 0001 P.falciparum FilterBloodSpot k13-crt-mdr-dhfr-dhps-cytB-cpmp-pfs47 \n",
    "\n",
    "                     - Pooled sequenced sample ID: 17GNDoxxx001P10F1290 = 2017 Guinea Dorota **xx x** 001 **Pooled SamplesInPool** P.falciparum FilterBloodSpot k13-crt-mdr-dhfr-dhps-cytB-cpmp-pfs47 \n",
    "\n",
    "                         - NOTE: If information is not availble (na) **x** is used. For pooled samples, DayofTreatment and Treatment is na since its a pool of multiple samples with that info. \n",
    "                         - NOTE: For pooled samples, **Genus** is replaced with **Pooled** and **SampleType** with **SamplesInPool** to indicated this as a pooled sequenced sample and sample count in each pool. \n",
    "         <p>&nbsp;</p>\n",
    "         - \"Year\" = the year the study was conducted \n",
    "         - \"Site\" = the state or province \n",
    "         - \"Day_of_treatment\" = describes the day of treatment provided to the patient \n",
    "         - \"Gene\" = drug resistant gene(s) \n",
    "         - \"G_annotation\" = full SNP annotation in the following format: WildTypeAA-CodonPosition-MutantAA \n",
    "         - \"Coverage\" = the number of reads covering the SNP \n",
    "         - \"VAF\" = variant allele frequency calculated by AA divided by total reads in loci \n",
    "         - \"SNP\" = single nucleotide polymorphism in WildTypeAA or MutantAA annotation format \n",
    "         - \"Type\" = describes if it is a wild type or mutant SNP \n",
    "\n",
    "     - Each observation is a row for each sample ID (patient ID) \n",
    " \n",
    " #### TODO\n",
    " \n",
    " #### Activity Name\n",
    " - [ ] Write doc.string at the beginning of the code\n",
    " - [ ] Write detailed description with comment for line by line\n",
    " - [ ] Make the code more simple and accurate\n",
    " - [ ] Follow zen of python\n",
    "    \n",
    " #### Completed Activity âœ“\n",
    " - [x] Created marked down at the beginning of the file for description"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "83bfdd5a-14d1-4bdd-bfac-48e4f7ac2261",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_463/704937234.py:3: DtypeWarning: Columns (16,19,24,26,28,30,31,32,33,34,35,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,57,58,59,60,61,62,64,65,66,67,68) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  Geneious_DF=pd.read_csv(\"Annotations.csv\")\n",
      "/tmp/ipykernel_463/704937234.py:15: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  Geneious_DF_N1[\"TrackerSNP\"]=Geneious_DF_N1[\"Amino Acid Change\"].astype(str).str[0]+Geneious_DF_N1[\"CDS Codon Number\"].astype(int).astype(str)+Geneious_DF_N1[\"Amino Acid Change\"].astype(str).str[-1]\n",
      "/tmp/ipykernel_463/704937234.py:78: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  Combination_filtered[\"SITE\"]=Combination_filtered.apply(site, axis=1)\n",
      "/tmp/ipykernel_463/704937234.py:79: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  Combination_filtered[\"TreatmentDay\"]=Combination_filtered.apply(TreatmentDay, axis=1)\n",
      "/tmp/ipykernel_463/704937234.py:80: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  Combination_filtered[\"Pooled\"]=Combination_filtered.apply(Pooled, axis=1)\n",
      "/tmp/ipykernel_463/704937234.py:81: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  Combination_filtered[\"Year\"]=Combination_filtered.apply(year, axis=1)\n",
      "/tmp/ipykernel_463/704937234.py:82: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  Combination_filtered[\"TYPE\"]=Combination_filtered.apply(type, axis=1)\n",
      "/tmp/ipykernel_463/704937234.py:83: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  Combination_filtered[\"SNP\"]=Combination_filtered.apply(SNP, axis=1)\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "Geneious_DF=pd.read_csv(\"Annotations.csv\")\n",
    "#print(Geneious_DF[\"Coverage\"])\n",
    "#print(Geneious_DF[\"Sequence Name\"])\n",
    "Geneious_DF_N1=Geneious_DF[(Geneious_DF['Type']=='Polymorphism') & (Geneious_DF['Amino Acid Change'].notnull())]\n",
    "\n",
    "Geneious_DF_N2=Geneious_DF[Geneious_DF['Type']=='Coverage - High']\n",
    "#Coverage - High\n",
    "\n",
    "#print(len(Geneious_DF_N1))\n",
    "#print(len(Geneious_DF_N2))\n",
    "#print(Geneious_DF_N1[\"CDS Codon Number\"])\n",
    "#print(Geneious_DF_N1[\"Amino Acid Change\"])\n",
    "Geneious_DF_N1[\"TrackerSNP\"]=Geneious_DF_N1[\"Amino Acid Change\"].astype(str).str[0]+Geneious_DF_N1[\"CDS Codon Number\"].astype(int).astype(str)+Geneious_DF_N1[\"Amino Acid Change\"].astype(str).str[-1]\n",
    "\n",
    "#Geneious_DF_N_fix=Geneious_DF_N1.loc[Geneious_DF_N1['Document Name'].str.contains('M05039', na=False)]\n",
    "#print(Geneious_DF_N1[\"TrackerSNP\"])\n",
    "#Geneious_DF_N1.to_csv(\"test.csv\", sep=',')\n",
    "#Geneious_DF_N1_test=pd.isna(Geneious_DF_N1[\"Amino Acid Change\"])\n",
    "#print(Geneious_DF_N1.iloc[1])\n",
    "#print(Geneious_DF_N1[\"Amino Acid Change\"])\n",
    "\n",
    "\n",
    "Combine_Variant_Wildtpye = [Geneious_DF_N1, Geneious_DF_N2]\n",
    "Combation_Vi_Wi = pd.concat(Combine_Variant_Wildtpye)\n",
    "#print(len(Combation_Vi_Wi))\n",
    "#print(len(Geneious_DF_N1))\n",
    "Combination_filtered=Combation_Vi_Wi.drop_duplicates(subset =[\"Document Name\", \"TrackerSNP\"] ) \n",
    "#(Geneious_DF_N1[!Geneious_DF_N1$feature %in% Geneious_DF_N2$feature, ]\n",
    "#\"Sample,Pooled,Year,SITE,TreatmentDay,GENE,G_annotation,COVERAGE,VAF,VF,SNP,TYPE\\n\")\n",
    "        \n",
    "def site(row):\n",
    "    if row['Document Name'][4:6]==\"Ha\":\n",
    "        return 'Hamdalaye'\n",
    "    elif row['Document Name'][4:6]==\"Do\":\n",
    "        return 'Dorota'\n",
    "    elif row['Document Name'][4:6]==\"Ma\":\n",
    "        return 'Maferinyah'\n",
    "    elif row['Document Name'][4:6]==\"La\":\n",
    "        return 'Lay-Sare'\n",
    "    elif row['Document Name'][4:6]==\"LS\":\n",
    "        return 'Lay-Sare'\n",
    "    \n",
    "def TreatmentDay(row):\n",
    "    if row['Document Name'][6:8]==\"00\":\n",
    "        return '0'\n",
    "    elif row['Document Name'][6:8]==\"1A\":\n",
    "        return '1'\n",
    "    elif row['Document Name'][6:8]!=\"00\" and row['Document Name'][6:8]!=\"1A\":\n",
    "        return row['Document Name'][6:8]\n",
    "    \n",
    "def Pooled(row):\n",
    "    if row['Document Name'][8:10]==\"xp\":\n",
    "        return 'individual'\n",
    "    elif row['Document Name'][8:10]!=\"xp\":\n",
    "        return 'pooled'\n",
    "\n",
    "def year(row):\n",
    "    return row['Document Name'][0:2]\n",
    "\n",
    "def type(row):\n",
    "    if row['Type'] =='Polymorphism':\n",
    "        return \"mutation\"\n",
    "    if row['Type'] =='Coverage - High':\n",
    "        return \"wildtype\"\n",
    "    \n",
    "def SNP(row):\n",
    "    if row['Type'] =='Polymorphism':\n",
    "        return row['TrackerSNP'][1::]\n",
    "    if row['Type'] =='Coverage - High':\n",
    "        return row['TrackerSNP'][0:-1]\n",
    "    \n",
    "    \n",
    "\n",
    "\n",
    "    \n",
    "Combination_filtered[\"SITE\"]=Combination_filtered.apply(site, axis=1)\n",
    "Combination_filtered[\"TreatmentDay\"]=Combination_filtered.apply(TreatmentDay, axis=1)\n",
    "Combination_filtered[\"Pooled\"]=Combination_filtered.apply(Pooled, axis=1)\n",
    "Combination_filtered[\"Year\"]=Combination_filtered.apply(year, axis=1)\n",
    "Combination_filtered[\"TYPE\"]=Combination_filtered.apply(type, axis=1)\n",
    "Combination_filtered[\"SNP\"]=Combination_filtered.apply(SNP, axis=1)\n",
    "\n",
    "#print(Combination_filtered[\"SITE\"])\n",
    "#print(Combination_filtered[\"TreatmentDay\"])\n",
    "#print(Combination_filtered[\"Pooled\"])\n",
    "#print(Combination_filtered[\"Year\"])\n",
    "#print(Combination_filtered[\"TYPE\"])\n",
    "\n",
    "\n",
    "#,\"SITE\",\"TreatmentDay\",\"Pooled\",\"Year\"\n",
    "\n",
    "Combination_report1=Combination_filtered[Combination_filtered['Type']=='Polymorphism']\n",
    "Combination_report2=Combination_filtered[Combination_filtered['Type']=='Coverage - High']\n",
    "final_report1=Combination_report1[[\"Document Name\",\"Sequence Name\",\"SITE\",\"TreatmentDay\",\"Pooled\",\"Year\",\"Coverage\",\"Variant Frequency\",\"Variant Raw Frequency\",\"TrackerSNP\",\"TYPE\",\"SNP\"]]\n",
    "final_report2=Combination_report2[[\"Document Name\",\"Sequence Name\",\"SITE\",\"TreatmentDay\",\"Pooled\",\"Year\",\"Average Coverage\",\"Variant Frequency\",\"Variant Raw Frequency\",\"TrackerSNP\",\"TYPE\",\"SNP\"]]\n",
    "final_report2_re=final_report2.rename(columns={'Average Coverage': 'Coverage'})\n",
    "#print(Combination_report)\n",
    "final_combine=[final_report1, final_report2_re]\n",
    "final_combine_2=pd.concat(final_combine)\n",
    "final_combine_2.to_csv(\"test.csv\", sep=',', index=False)\n",
    "#print((Combation_Vi_Wi))\n",
    "#print(len(Geneious_DF_N2))\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "bc46cb9f-c06f-435b-96cb-45ebd104f1b3",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_463/468509530.py:27: FutureWarning: In a future version of pandas all arguments of DataFrame.drop except for the argument 'labels' will be keyword-only.\n",
      "  df_merged_poolsize = df_merged_poolsize.drop('SITE_y', 1)\n",
      "/tmp/ipykernel_463/468509530.py:28: FutureWarning: In a future version of pandas all arguments of DataFrame.drop except for the argument 'labels' will be keyword-only.\n",
      "  df_merged_poolsize = df_merged_poolsize.drop('YEAR', 1)\n"
     ]
    }
   ],
   "source": [
    "#print(final_combine_2[\"Document Name\"])\n",
    "\n",
    "pooled_part1=pd.read_csv(\"Pooled_Info_Part1.csv\")\n",
    "pooled_part2=pd.read_csv(\"Pooled_Info_Part2_fixed.csv\")\n",
    "\n",
    "Combine_pooled_parts = [pooled_part1[[\"Pool\",\"SITE\",\"YEAR\",\"AMD_ID\",\"Poolsize\"]], pooled_part2[[\"Pool\",\"SITE\",\"YEAR\",\"AMD_ID\",\"Poolsize\"]]]\n",
    "Combation_pooled_concatenate = pd.concat(Combine_pooled_parts)\n",
    "\n",
    "#print(Combation_pooled_concatenate)\n",
    "Combation_pooled_concatenate.to_csv(\"test-pre1.csv\", sep=',', index=False)\n",
    "\n",
    "#print(pooled_part1[\"Poolsize \"])\n",
    "s = pd.Series(pooled_part1[\"AMD_ID\"])\n",
    "s2 = pd.Series(pooled_part1[\"Poolsize\"])\n",
    "s3 = pd.Series(pooled_part1[\"Pool\"])\n",
    "\n",
    "def name(row):\n",
    "    return row['Document Name'].split(\"_\")[0]\n",
    "\n",
    "final_combine_2[\"Document Name\"]=final_combine_2.apply(name, axis=1)\n",
    "\n",
    "final_combine_2.rename(columns={'Document Name':'AMD_ID'}, inplace=True)\n",
    "\n",
    "\n",
    "df_merged_poolsize = pd.merge(final_combine_2, Combation_pooled_concatenate, on=['AMD_ID'], how='left') \n",
    "\n",
    "df_merged_poolsize = df_merged_poolsize.drop('SITE_y', 1)\n",
    "df_merged_poolsize = df_merged_poolsize.drop('YEAR', 1)\n",
    "\n",
    "#final_combine_2.loc[(final_combine_2['Document Name']==Combation_pooled_concatenate[\"AMD_ID\"])] = Combation_pooled_concatenate[\"Poolsize\"]\n",
    "\n",
    "#print(final_combine_2[\"Document Name\"])\n",
    "#df[df.Name.isin(['Alice', 'Bob'])]\n",
    "\n",
    "#def row_match(row):\n",
    "#    return Combation_pooled_concatenate[\"AMD_ID\"].isin([row['Document Name']])\n",
    "      \n",
    "#new_match=(final_combine_2.apply(row_match, axis=1))\n",
    "#print(new_match)\n",
    "\n",
    "#print(new_match.columns)\n",
    "\n",
    "#print(new_match.iloc[0][new_match.iloc[0]==True].index.tolist())\n",
    "\n",
    "df_merged_poolsize.Poolsize.fillna(value=1, inplace=True)\n",
    "\n",
    "df_merged_poolsize.to_csv(\"test2.csv\", sep=',', index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fbdafdd2-4e85-441f-a444-7ae3318ad7a3",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
