{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "f966b203-17ec-42c8-bf41-f337bdc5bf5b",
   "metadata": {},
   "source": [
    "## Geneious analysis for individual samples from raw Geneious output, \"Annotation.csv\" \n",
    "\n",
    " ### Required packages\n",
    " - No specific package required \n",
    " \n",
    " ### Inputs\n",
    " - Geneious SNP analysis of _k13_, _crt_, _mdr1_, _dhfr_, _dhps_, and _cytb_\n",
    " - Documentation on Geneious analysis can be found: Readme.md\n",
    " - Geneious outputs were modified to GuineaAnalysis_Individual.csv from \"Annotation.csv\"\n",
    " \n",
    " \n",
    " ### Data structure \n",
    " - [Long-form](https://seaborn.pydata.org/tutorial/data_structure.html#long-form-vs-wide-form-data) \n",
    "     - Each variable is a column \n",
    "\n",
    "         - \"Sample\" = *AMD ID*, including associated meta-data for each sample\n",
    "             - AMD ID and bit code key is found under MS Teams > Domestic > Files > Sample Naming > Sample_naming_key.pptx  \n",
    "\n",
    "             - Key: **Year Country State/Site DayofTreatment Treatment SampleID Genus SampleType GeneMarker-8bitcode SampleSeqCount**\n",
    "\n",
    "                 - Example:\n",
    "                     - Individual sequenced sample ID: 17GNDo00F0001PfF1290 = 2017 Guinea Dorota Day0 AS+AQ 0001 P.falciparum FilterBloodSpot k13-crt-mdr-dhfr-dhps-cytB-cpmp-pfs47 \n",
    "\n",
    "                     - Pooled sequenced sample ID: 17GNDoxxx001P10F1290 = 2017 Guinea Dorota **xx x** 001 **Pooled SamplesInPool** P.falciparum FilterBloodSpot k13-crt-mdr-dhfr-dhps-cytB-cpmp-pfs47 \n",
    "\n",
    "                         - NOTE: If information is not availble (na) **x** is used. For pooled samples, DayofTreatment and Treatment is na since its a pool of multiple samples with that info. \n",
    "                         - NOTE: For pooled samples, **Genus** is replaced with **Pooled** and **SampleType** with **SamplesInPool** to indicated this as a pooled sequenced sample and sample count in each pool. \n",
    "         <p>&nbsp;</p>\n",
    "         - \"Year\" = the year the study was conducted \n",
    "         - \"Site\" = the state or province \n",
    "         - \"Day_of_treatment\" = describes the day of treatment provided to the patient \n",
    "         - \"Gene\" = drug resistant gene(s) \n",
    "         - \"G_annotation\" = full SNP annotation in the following format: WildTypeAA-CodonPosition-MutantAA \n",
    "         - \"Coverage\" = the number of reads covering the SNP \n",
    "         - \"VAF\" = variant allele frequency calculated by AA divided by total reads in loci \n",
    "         - \"SNP\" = single nucleotide polymorphism in WildTypeAA or MutantAA annotation format \n",
    "         - \"Type\" = describes if it is a wild type or mutant SNP \n",
    "\n",
    "     - Each observation is a row for each sample ID (patient ID) \n",
    " \n",
    " #### TODO\n",
    " \n",
    " #### Activity Name\n",
    " - [ ] Write doc.string at the beginning of the code\n",
    " - [ ] Write detailed description with comment for line by line\n",
    " - [ ] Make the code more simple and accurate\n",
    " - [ ] Follow zen of python\n",
    "    \n",
    " #### Completed Activity âœ“\n",
    " - [x] Created marked down at the beginning of the file for description"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "c3651295-d9a7-476b-a2e7-43c0b98bceb7",
   "metadata": {},
   "outputs": [],
   "source": [
    "#######This document is for individual samples report for the EPI team#######\n",
    "\n",
    "\"\"\"\n",
    "This is individual_geneious_EP jupyter notebook file. This file is for processing raw Geneious output file \n",
    "\"Annotation.csv\" file into GuineaAnalysis_Individual.csv file which contains information filtered such as \n",
    "Sample,Pooled,Year,SITE,TreatmentDay,GENE,G_annotation,COVERAGE,VAF,VF,SNP,TYPE. Sample means AMD_ID which\n",
    "has naming schema including other informations within the name. Pooled means whether the sample is an individual\n",
    "sample or pooled sample. This jupyter notebook first assigns variants. Then, it checks for silent mutations. \n",
    "At last it adds full wildtypes. The differnce from Lab individual jupyter notebook is that this file uses one\n",
    "additional dictionary to filter out SNPs that are not reportable.\n",
    "\"\"\"\n",
    "\n",
    "######First I create a dictionary of reportable SNPs so that I could later use it to filter out unreportable SNP samples####\n",
    "dict_reportables_filter={}  ##Create a dictionary for filtering reportable SNPs\n",
    "#####The voinew3.csv file contains a table of reportable SNPs########\n",
    "with open (\"voinew3.csv\", \"r\") as v1:  ##voinew3.csv file contains information reportable SNPs \n",
    "    ####I loop through lines in the voi file ###\n",
    "    for lines in v1:  \n",
    "        ###I would like to skip the header####\n",
    "        if \"Ref\" not in lines:\n",
    "            ######I add Gene name and SNP####\n",
    "            dict_reportables_filter[lines.split(\",\")[2]+lines.split(\",\")[3]+lines.split(\",\")[4].strip(\"\\n\")]=\"exist\"\n",
    "\n",
    "###Open raw output table file \"Annotations.csv\" from Geneious#####\n",
    "import itertools\n",
    "###### Assign letters to the amino acid names #########\n",
    "#####This is a dictionary assigning three letters to amino acid letter #####\n",
    "AAdic = {  ###Amino Acid dictionary which assigns an amino acid letters to three letters of amino acid names####\n",
    "\"Ala\": \"A\",\n",
    "\"Arg\": \"R\",\n",
    "\"Asn\": \"N\",\n",
    "\"Asp\": \"D\",\n",
    "\"Asx\": \"B\",\n",
    "\"Cys\": \"C\",\n",
    "\"Glu\": \"E\",\n",
    "\"Gln\": \"Q\",\n",
    "\"Glx\": \"Z\",\n",
    "\"Gly\": \"G\",\n",
    "\"His\": \"H\",\n",
    "\"Ile\": \"I\",\n",
    "\"Leu\": \"L\",\n",
    "\"Lys\": \"K\",\n",
    "\"Met\": \"M\",\n",
    "\"Phe\": \"F\",\n",
    "\"Pro\": \"P\",\n",
    "\"Ser\": \"S\",\n",
    "\"Thr\": \"T\",\n",
    "\"Trp\": \"W\",\n",
    "\"Tyr\": \"Y\",\n",
    "\"Val\": \"V\",\n",
    "\"STOP\": \"X\",\n",
    "}\n",
    "#######Convert combination of bases to a codon############\n",
    "######This is coverting three base combination into a coddon####\n",
    "#####It would create dicionary of three bases conversion to codons######\n",
    "Current_Codon = \"\"\n",
    "Codon_Combinations = [\"\".join(i) for i in itertools.product([\"T\", \"G\", \"A\", \"C\"], repeat=3)]  ####Create combination of\n",
    "### 4 letters into three letters\n",
    "Codon_Dic = {}  ### Create a dictionary for codon\n",
    "for x in Codon_Combinations:  ###Loop over combinations\n",
    "    Current_Codon = \"\"   ###At beginning of loop current codon is assigned as empty string\n",
    "    for y in x:  ###For letter in combination\n",
    "        Current_Codon += y  ###Add each letter from the combination to form the current codon\n",
    "    if Current_Codon == \"TTT\" or Current_Codon == \"TTC\":\n",
    "        Codon_Dic[Current_Codon] = \"Phe\"\n",
    "    if Current_Codon == \"TTA\" or Current_Codon == \"TTG\":\n",
    "        Codon_Dic[Current_Codon] = \"Leu\"\n",
    "    if (\n",
    "        Current_Codon == \"CTT\"\n",
    "        or Current_Codon == \"CTC\"\n",
    "        or Current_Codon == \"CTA\"\n",
    "        or Current_Codon == \"CTG\"\n",
    "    ):\n",
    "        Codon_Dic[Current_Codon] = \"Leu\"\n",
    "    if Current_Codon == \"ATT\" or Current_Codon == \"ATC\" or Current_Codon == \"ATA\":\n",
    "        Codon_Dic[Current_Codon] = \"Ile\"\n",
    "    if Current_Codon == \"ATG\":\n",
    "        Codon_Dic[Current_Codon] = \"Met\"\n",
    "    if (\n",
    "        Current_Codon == \"GTT\"\n",
    "        or Current_Codon == \"GTC\"\n",
    "        or Current_Codon == \"GTA\"\n",
    "        or Current_Codon == \"GTG\"\n",
    "    ):\n",
    "        Codon_Dic[Current_Codon] = \"Val\"\n",
    "    if (\n",
    "        Current_Codon == \"TCT\"\n",
    "        or Current_Codon == \"TCC\"\n",
    "        or Current_Codon == \"TCA\"\n",
    "        or Current_Codon == \"TCG\"\n",
    "    ):\n",
    "        Codon_Dic[Current_Codon] = \"Ser\"\n",
    "    if (\n",
    "        Current_Codon == \"CCT\"\n",
    "        or Current_Codon == \"CCC\"\n",
    "        or Current_Codon == \"CCA\"\n",
    "        or Current_Codon == \"CCG\"\n",
    "    ):\n",
    "        Codon_Dic[Current_Codon] = \"Pro\"\n",
    "    if (\n",
    "        Current_Codon == \"ACT\"\n",
    "        or Current_Codon == \"ACC\"\n",
    "        or Current_Codon == \"ACA\"\n",
    "        or Current_Codon == \"ACG\"\n",
    "    ):\n",
    "        Codon_Dic[Current_Codon] = \"Thr\"\n",
    "    if (\n",
    "        Current_Codon == \"GCT\"\n",
    "        or Current_Codon == \"GCC\"\n",
    "        or Current_Codon == \"GCA\"\n",
    "        or Current_Codon == \"GCG\"\n",
    "    ):\n",
    "        Codon_Dic[Current_Codon] = \"Ala\"\n",
    "    if Current_Codon == \"TAT\" or Current_Codon == \"TAC\":\n",
    "        Codon_Dic[Current_Codon] = \"Tyr\"\n",
    "    if Current_Codon == \"CAT\" or Current_Codon == \"CAC\":\n",
    "        Codon_Dic[Current_Codon] = \"His\"\n",
    "    if Current_Codon == \"CAA\" or Current_Codon == \"CAG\":\n",
    "        Codon_Dic[Current_Codon] = \"Gln\"\n",
    "    if Current_Codon == \"AAT\" or Current_Codon == \"AAC\":\n",
    "        Codon_Dic[Current_Codon] = \"Asn\"\n",
    "    if Current_Codon == \"AAA\" or Current_Codon == \"AAG\":\n",
    "        Codon_Dic[Current_Codon] = \"Lys\"\n",
    "    if Current_Codon == \"GAT\" or Current_Codon == \"GAC\":\n",
    "        Codon_Dic[Current_Codon] = \"Asp\"\n",
    "    if Current_Codon == \"GAA\" or Current_Codon == \"GAG\":\n",
    "        Codon_Dic[Current_Codon] = \"Glu\"\n",
    "    if Current_Codon == \"TGT\" or Current_Codon == \"TGC\":\n",
    "        Codon_Dic[Current_Codon] = \"Cys\"\n",
    "    if Current_Codon == \"TGG\":\n",
    "        Codon_Dic[Current_Codon] = \"Trp\"\n",
    "    if (\n",
    "        Current_Codon == \"AGA\"\n",
    "        or Current_Codon == \"AGG\"\n",
    "        or Current_Codon == \"CGT\"\n",
    "        or Current_Codon == \"CGC\"\n",
    "        or Current_Codon == \"CGA\"\n",
    "        or Current_Codon == \"CGG\"\n",
    "    ):\n",
    "        Codon_Dic[Current_Codon] = \"Arg\"\n",
    "    if Current_Codon == \"AGT\" or Current_Codon == \"AGC\":\n",
    "        Codon_Dic[Current_Codon] = \"Ser\"\n",
    "    if (\n",
    "        Current_Codon == \"GGT\"\n",
    "        or Current_Codon == \"GGC\"\n",
    "        or Current_Codon == \"GGA\"\n",
    "        or Current_Codon == \"GGG\"\n",
    "    ):\n",
    "        Codon_Dic[Current_Codon] = \"Gly\"\n",
    "    if Current_Codon == \"TGA\" or Current_Codon == \"TAA\" or Current_Codon == \"TAG\":\n",
    "        Codon_Dic[Current_Codon] = \"STOP\"\n",
    "        \n",
    "filter_exist_dict={}##Create a dictionary to filter out duplicate information. For example when there is information about variant for a sample, meaning variant exsits,\n",
    "##Then we do not need to consider having full wildtypes\n",
    "#####Open Geneious raw output file which is Annotation.csv####\n",
    "####This is the ulfiltered file######\n",
    "with open (\"Annotations.csv\", \"r\") as r1:  ###Import raw geneious output for data cleaning and analysis purpose\n",
    "    ####We will write filterd information on GuineaAnalysis_individual_EP######\n",
    "    with open(\"GuineaAnalysis_Individual_EP.csv\" , \"w\") as w1:\n",
    "        ####We will write the header first with given information#####\n",
    "        w1.write(\"Sample,Pooled,Year,SITE,TreatmentDay,GENE,G_annotation,COVERAGE,VAF,VF,SNP,TYPE\\n\")  ##Write columns for the information for the cleaned data\n",
    "        ####We will create a dictionary with area code in it so that####\n",
    "        ####We can infer name of region from the sample ID#####\n",
    "        dict1={}\n",
    "        dict1[\"Ha\"]=\"Hamdalaye\"\n",
    "        dict1[\"Do\"]=\"Dorota\"\n",
    "        dict1[\"Ma\"]=\"Maferinyah\"\n",
    "        dict1[\"La\"]=\"Lay-SarÃ¨\"\n",
    "        dict1[\"LS\"]=\"Lay-SarÃ¨\"\n",
    "        dict2={}\n",
    "        #####We will loop through each line in the raw Geneious output file######\n",
    "        for lines in r1:   ##run a for loop for each lines in Annotation.csv to analyze line by line\n",
    "            ######Check if there is Polymorphism in the line from the table#######\n",
    "            ######If there is polymorphism then that means there is variant######\n",
    "            ######According to Geneious as geneious report variants as polymorphisms####\n",
    "            ######We will loop through Geneious raw output for variants first####\n",
    "            if \"Polymorphism\" in lines and \"Variants\" in lines:  ##Check if two words, polymorphism and variants, are in the lines. These words indicate\n",
    "                ##There are variants of interest in the line, Thus, we would like to extract the information the given line\n",
    "                ####sample name is first column right before the _ string#####\n",
    "                ####We will filer out unnecessary information from sample name#####\n",
    "                samplename=lines.split(\",\")[0].split(\"_\")[0]  ##We get sample name which is first part before string \"_\"\n",
    "                ####Sample gene is the second coulmmn in the line####\n",
    "                ###We will assign genes to the given variants of interest####\n",
    "                samplegene=lines.split(\",\")[1]  ##The gene is found in the second column split by \",\"\n",
    "                ###We want to correct name for DHPS####\n",
    "                if \"DHPS\" in samplegene:  ##Fix name for DHPS\n",
    "                    samplegene=\"DHPS\"\n",
    "                ###We want to correct name of MT CYTB####\n",
    "                if \"mitochondrial\" in samplegene:  ##Fix name for cytob\n",
    "                    samplegene=\"CYTB\"\n",
    "                ###Sample AA change is column 17 which has the amino acid changes####\n",
    "                sampleAAchange=lines.split(\",\")[16]   ##We identify AA change which is 17th column\n",
    "                ####If there is Amino acid change reported combine it with the c variable which is codon position####\n",
    "                ####to create a SNP change report #####\n",
    "                if sampleAAchange!=\"\":  ##If we find AA change at 17th column we want to use that information to build SNP\n",
    "                    ####Based on spacing you could figurout aminoacid letters for before changing after changing#####\n",
    "                    a=sampleAAchange.split(\" \")[0]  ##Wildtype amino acid\n",
    "                    b=sampleAAchange.split(\" \")[2]  ##mutation amino acid\n",
    "                    ####C is the aminoacid position where change has happened######\n",
    "                    c=lines.split(\",\")[20]  ##amino acid  position\n",
    "                    AAchangereport=(a+c+b)  ## If you add a+c+b then SNP is obtained\n",
    "                    ###Next We will assign variant allele frequency to a column####\n",
    "                    sampleVAF=(lines.split(\"M05039\")[0]).split(\",\")[-4]  ##Four coulms before M05039 finds the variant allele frequency\n",
    "                    ###Next We will assign variant frequency to the next column####\n",
    "                    sampleVF=(lines.split(\"M05039\")[0]).split(\",\")[-2]   ##Two columns before M05039 finds the variant frequency\n",
    "                    ####We want to assign coverage information to the next column####\n",
    "                    sampleCoverage=lines.split(\",\")[28]   ##The coverage of sample is found on 29th column\n",
    "                    ###We want to make sure these are Guinea sample which has GN in its name####\n",
    "                    if \"GN\" in samplename:  ##Check if this is guinea sample and not a control\n",
    "                        ####Use a dictionary to prevent overlaps######\n",
    "                        ####We want to use condition that there shouldn't be letter p in that position#####\n",
    "                        ####To identify individual samples also we don't want to process duplicates such as ####\n",
    "                        ####duplicate wildtypes or variants once we processed them because there can be only either variant or wildtype####\n",
    "                        ####Also for EPI report the SNP has to be in the previously created dictionary#####\n",
    "                        ####We want to make the dictionary without the last letter which is changed amino acid because ####\n",
    "                        ####We want to prevent situation such as same amino acid change and same wildtype but different mutation change #####\n",
    "                        if samplename[9]!=\"p\" and (samplename,AAchangereport[0:-1]) not in filter_exist_dict and AAchangereport in dict_reportables_filter:  ##Check if this sample is not pooled\n",
    "                            ##In addition check if it is a reportable SNPs\n",
    "                            ####We want to assign to dictionary3 for preventing duplicates in the future####\n",
    "                            filter_exist_dict[samplename,AAchangereport[0:-1]]=\"exist\"   ##Assign a dictionary so that we won't repeat assigning to create duplicates\n",
    "                            #####If the 1A is in that postion of sample ID we assign treatment day 1#####\n",
    "                            #####We assign information based on the header we wrote previously#####\n",
    "                            if samplename[6:8]==\"1A\":  ##Check treatment day that if it is 1A in given position it means that it was treated on the first day\n",
    "                                w1.write(samplename+\",\"+\"Individual\"+\",\"+samplename[0:2]+\",\"+dict1[samplename[4:6]]+\",\"+\"01\"+\",\"+samplegene+\",\"+AAchangereport+\",\"+sampleCoverage+\",\"+sampleVAF+\",\"+sampleVF+\",\"+AAchangereport[1::]+\",\"+\"mutation\"+\"\\n\")\n",
    "                            ####Otherwise we assign the treatment day based on name####\n",
    "                            else:  ##Else we assign treatment day based on given position of the sample name\n",
    "                                w1.write(samplename+\",\"+\"Individual\"+\",\"+samplename[0:2]+\",\"+dict1[samplename[4:6]]+\",\"+samplename[6:8]+\",\"+samplegene+\",\"+AAchangereport+\",\"+sampleCoverage+\",\"+sampleVAF+\",\"+sampleVF+\",\"+AAchangereport[1::]+\",\"+\"mutation\"+\"\\n\")\n",
    "                #####For silent mutation when there is no amino acid change####\n",
    "                #####We check if there is mutation change or not####\n",
    "                ####If there is no amino acid change we considered it silent mutation####\n",
    "                if sampleAAchange==\"\" and lines.split(\",\")[26]!=\"\":  ##This part is for assigning silent mutation\n",
    "                    ####For silent mutation we have to use three base changes instead of aminoacid letter change###\n",
    "                    ####Due to how Geneious reports it#####\n",
    "                    ####We use the previously created codon dictionary and amino acid dictionary to get the amino acid letter for the three bases changes####\n",
    "                    a=AAdic[Codon_Dic[lines.split(\",\")[26][0:3]]]  ##Assign a amino acid using codon dictionary and amino acid dictionary createed previously\n",
    "                    ##This is the wildtype part of the SNP\n",
    "                    b=AAdic[Codon_Dic[lines.split(\",\")[26][-3::]]]  ##Assign a mutation part of the SNP using codon dictionary and amino acid dictionary \n",
    "                    ####C is the location of amino acid change####\n",
    "                    c=lines.split(\",\")[20]  ##Assign a amino acid location for the given SNP\n",
    "                    AAchangereport=(a+c+b)\n",
    "                    ####Again we get the sample variant allele frequency and variant frequency depending on the location in the geneious raw output table#### \n",
    "                    sampleVAF=(lines.split(\"M05039\")[0]).split(\",\")[-4]  ##Four coulms before M05039 finds the variant allele frequency\n",
    "                    sampleVF=(lines.split(\"M05039\")[0]).split(\",\")[-2]  ##Two columns before M05039 finds the variant frequency\n",
    "                    ####We also get the average coverage depednign on which column it is###\n",
    "                    ####I need to further investigate using average coverage vs maximum coverage####\n",
    "                    sampleCoverage=lines.split(\",\")[28]   ##The coverage of sample is found on 29th column\n",
    "                        ####Use a dictionary to prevent overlaps######\n",
    "                    if samplename[9]!=\"p\" and (samplename,AAchangereport[0:-1]) not in filter_exist_dict and AAchangereport in dict_reportables_filter:  ##Check if this sample is not pooled\n",
    "                        ##In addition check if it is a reportable SNPs\n",
    "                        ####We want to assign to dictionary3 for preventing duplicates in the future####\n",
    "                        filter_exist_dict[samplename,AAchangereport[0:-1]]=\"exist\"  ##Prevent duplicates information\n",
    "                        #####If the 1A is in that postion of sample ID we assign treatment day 1#####\n",
    "                        #####We assign information based on the header we wrote previously#####\n",
    "                        if samplename[6:8]==\"1A\":  ##Check for the treatment day of 1\n",
    "                            w1.write(samplename+\",\"+\"Individual\"+\",\"+samplename[0:2]+\",\"+dict1[samplename[4:6]]+\",\"+samplename[6:8]+\",\"+samplegene+\",\"+AAchangereport+\",\"+sampleCoverage+\",\"+sampleVAF+\",\"+sampleVF+\",\"+AAchangereport+\",\"+\"silent\"+\"\\n\")\n",
    "                        ####Otherwise we assign the treatment day based on name####\n",
    "                        else:  ##Check for the treatment day later than 1\n",
    "                            w1.write(samplename+\",\"+\"Individual\"+\",\"+samplename[0:2]+\",\"+dict1[samplename[4:6]]+\",\"+samplename[6:8]+\",\"+samplegene+\",\"+AAchangereport+\",\"+sampleCoverage+\",\"+sampleVAF+\",\"+sampleVF+\",\"+AAchangereport+\",\"+\"silent\"+\"\\n\")\n",
    "\n",
    "####We open the annotation file again for identifying pure wildtpye this time####\n",
    "with open (\"Annotations.csv\", \"r\") as r1:   ##Import geneious raw output again\n",
    "    with open(\"GuineaAnalysis_Individual_EP.csv\" , \"a\") as w1:  ##Add full widtype information to the table\n",
    "        ####We go through each line in the annotation file####\n",
    "        for lines in r1:\n",
    "            ####This time conditional would be only for coverage####\n",
    "            ####If there is coverage and it is not 0 that means####\n",
    "            ####It is wildtype#####\n",
    "            if \"Coverage:\" in lines:  ##If coverage exists in the line we consider it wildtype if that is not variant\n",
    "                ####sample name is first column right before the _ string#####\n",
    "                ####We will filer out unnecessary information from sample name#####\n",
    "                samplename=lines.split(\",\")[0].split(\"_\")[0]\n",
    "                ####Sample gene is the second coulmmn in the line####\n",
    "                ###We will assign genes to the given wildtypes of interest####\n",
    "                samplegene=lines.split(\",\")[1]\n",
    "                ###We want to correct name for DHPS####\n",
    "                if \"DHPS\" in samplegene:  ##correct dhps name\n",
    "                    samplegene=\"DHPS\"\n",
    "                ###We want to correct name of MT CYTB####\n",
    "                if \"mitochondrial\" in samplegene:  ##correct my cytob name\n",
    "                    samplegene=\"CYTB\"\n",
    "                ####For lines containing Coverage instead of polymorphism###\n",
    "                ###The aminoacid change is reported at different column####\n",
    "                ###It is reported at the last column as trackerSNP####\n",
    "                AAchangereport=lines[0:-12].split(\",\")[-1]  ##check SNP\n",
    "                ###Since this is wildtype sampleVAF would be 0%####\n",
    "                ###Sample variant frequency is also 0#####\n",
    "                sampleVAF=\"0%\"  ##sample VAF is 0% because this is full wildtype\n",
    "                sampleVF=\"0\"  ##sample variant frequency is 0 because this is full wildtype\n",
    "                ####For lines containing Coverage instead of polymorphism###\n",
    "                ###The sample coverage is reported at different column####\n",
    "                sampleCoverage=lines.split(\",\")[17]  ##assign sample coverage\n",
    "                ####Again we are only interested in Guinea samples####\n",
    "                if \"GN\" in samplename:  ##check if this is guinea sample\n",
    "                    ####Since this is individual report for EPI team we are gonna find individuals that are not p####\n",
    "                    ####at a given position of the naming schema###\n",
    "                    if samplename[9]!=\"p\" and (samplename,AAchangereport[0:-1]) not in filter_exist_dict and AAchangereport in dict_reportables_filter:  ##Check if this sample is not pooled\n",
    "                        ##In addition check if it is a reportable SNPs\n",
    "                        ###We want to setup a dictionary so that we wouldn't create duplicate report from the variant report made previously#####\n",
    "                        filter_exist_dict[samplename,AAchangereport[0:-1]]=\"exist\"  ##check if variant sample is already assigned before assigning full wildtype\n",
    "                        ####If 1A is in the naming schema we treat it as treatment day of 1######\n",
    "                        if samplename[6:8]==\"1A\":   ##check treatment day\n",
    "                            w1.write(samplename+\",\"+\"Individual\"+\",\"+samplename[0:2]+\",\"+dict1[samplename[4:6]]+\",\"+\"01\"+\",\"+samplegene+\",\"+AAchangereport+\",\"+sampleCoverage+\",\"+sampleVAF+\",\"+sampleVF+\",\"+AAchangereport[0:-1]+\",\"+\"wildtype\"+\"\\n\")\n",
    "                        ####Otherwise we just assign the given treatment day in the naming schema#####\n",
    "                        else:\n",
    "                            w1.write(samplename+\",\"+\"Individual\"+\",\"+samplename[0:2]+\",\"+dict1[samplename[4:6]]+\",\"+samplename[6:8]+\",\"+samplegene+\",\"+AAchangereport+\",\"+sampleCoverage+\",\"+sampleVAF+\",\"+sampleVF+\",\"+AAchangereport[0:-1]+\",\"+\"wildtype\"+\"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e3b06e56-9c00-4683-8a0c-c69638ba0503",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
