{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "aca6cfe9-9b20-49cb-800f-30b351aa5cd3",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "Malaria has been a major illness in Africa. The drug resistance of malaria is making it more difficult to treat the patients with Malaria. To overcome the drug resistance CDC is developing a surveillance protocol for tracking drug ressitance of malarial genes. As part of the protocol and basic step we analyze Plasmodium Falciparum genes gained for patients. This ReadME is for other users to understand the work I've done which analyzes genes and reports findings. \n",
    "\n",
    "## ReadMe for Geneious workflow and scripts for reports for lab and EPI\n",
    "####  1. Geneious Workflow\n",
    "    Geneious Workflow contains mainly four steps for processing raw sequences to variants and coverage. You select a samples then go to workflows and run the SNP_Coverage_Analysis.geneiousWorkflow. Then, it automatically runs all the below steps from a -> d. \n",
    "    \n",
    "###### a. Trimming (Discard unnecessary reads that would cause noise to the data)\n",
    "    Geneious uses BBduk for trimming raw sequences. The settings are\n",
    "    1. Trim Adapters: All Truseq, Nextera and PhilX adapters \n",
    "    ✓ Trim Right End\n",
    "    ✓ Kmer Length 27 \n",
    "    ✓ Maximum Substitutions 1 \n",
    "    ✓ Maximum Substitutions + INDELs: 0,\n",
    "    2. Trim Low quality: \n",
    "    ✓Both Ends \n",
    "    ✓Minimum Quality 35\n",
    "    3. Trim Adapters based on paired read overhangs: Minimum Overlap: 24\n",
    "    4. Discard Short Reads: Minimum Length: 150 bp\n",
    "\n",
    "###### b. Map to Reference (To observe differences in genetic structure from reference)\n",
    "    The workflow uses Map to reference for algining reads to the reference\n",
    "    1. Dissolve contigs and re-assemble: reference sequecne is updated_MaRS_refs_ET (6sequences), \n",
    "    Assemble each sequence list separately\n",
    "    2. Mapper: \n",
    "    ✓ Bowtie2 \n",
    "    ✓ alginment Type: End to End \n",
    "    ✓ Use Preset: High Sensitivity/Medium\n",
    "    3. Do not trim (discrad trim annotations)\n",
    "    4. Save contigs\n",
    "    \n",
    "###### c. Find Variations/SNPs (To find the unique genetic structure from the samples)\n",
    "    1. Find Variants: Minimum Variant Frequency: 0.05\n",
    "    2. Analtze effect of variants on translations : Default Genetic Code, Standard\n",
    "    3. Advanced: Use separate annotations for each variant at a position\n",
    "     ✓ Record names of all contributing sequences of each variant\n",
    "     ✓ Only find variations in annotation types = TrackerSNP\n",
    "     ✓ Also find varations within 0bases of those types\n",
    "     ✓ CDS properties to Copy: gene, product, protein_id, locus, tag, note.\n",
    "    \n",
    "###### d. Find coverage (To see if there are WildTypes we are interested )\n",
    "    1. Only Find in: Annotations in reference sequence of type: TrackerSNP, Create coverage annotations on reference sequence\n",
    "    2. High Coverage: Find regions with coverage above, number of sequences: 0\n",
    "    \n",
    "###### e. Selecting workflow outputs (contigs )\n",
    "    1. After the workflow is done contig documents will be generated in the same folder with AMD_ID and gene information\n",
    "    2. To select the contigs you click on modified \n",
    "    * If you can't see modified header scroll to the right \n",
    "    3. Then select the files that are generated later with AMD_ID and gene information in the document title\n",
    "    \n",
    "###### f. Export findings to a csv raw output (To process more data )\n",
    "    1. Select all the documents with coverage and variant information\n",
    "    * Select in as mentioned in step e\n",
    "    2. Select all the columns\n",
    "    3. Select all the types\n",
    "    4. Export table \n",
    "    * Export as Annotations.csv\n",
    "\n",
    "\n",
    "#### 2. Python scripts for Geneious raw input\n",
    "###### a. Read Annotations.csv file and process\n",
    "    1. Python script has with open function to open file and read each lines from the file\n",
    "    2. Each jupyter note script will process Geneious raw output for different purposes.\n",
    "\n",
    "###### b. File1: Individual_geneious_EP/LAB (Individual and Pooled)\n",
    "    1 This jupyter note file is for processing individual file\n",
    "    2. This file produces the file with these headers: \n",
    "    * Sample,Pooled,Year,SITE,TreatmentDay,GENE,G_annotation,COVERAGE,VAF,VF,SNP,TYPE\n",
    "    3. The information in the header are obtained from running this jupyter file\n",
    "    4. The raw Geneious output will be transformed to .csv file with above header information\n",
    "    5. EP/LAB means there are two vesions of this code. EP stands for epidemiology team which has reportable mutatns while LAB version has other mutations.\n",
    "    \n",
    "###### c. File2: Pooled_geneious_EP/LAB (Pooled)\n",
    "    1 This jupyter note file is for processing Pooled file\n",
    "    2. This file produces the file with these headers: \n",
    "    * Sample,Pooled,Year,SITE,TreatmentDay,GENE,G_annotation,COVERAGE,VAF,VF,SNP,TYPE\n",
    "    3. The information in the header are obtained from running this jupyter file\n",
    "    4. The raw Geneious output will be transformed to .csv file with above header information\n",
    "    5. EP/LAB means there are two vesions of this code. EP stands for epidemiology team which has reportable mutatns while LAB version has other mutations.\n",
    "\n",
    "###### d. File3: geneiousanalysis1_combined1.ipynb (Individual and Pooled)\n",
    "    1. This jupyter note file is for processing combined pooled and individual files\n",
    "    2. The purpose of processing both pooled and individual file is for later producing\n",
    "    weighted average variant frequency file with total number of samples in the file.\n",
    "    3. This file produces the file with these headers: \n",
    "    * Sample,Pooled,Year,SITE,TreatmentDay,GENE,G_annotation,COVERAGE,VAF,VF,SNP,TYPE\n",
    "    4. The information in the header are obtained from the jupyter file\n",
    "    \n",
    "###### e. File4: Pooled_geneious_EP/LAB.ipynb\n",
    "    1. This jupyter note file is for processing adding pooled group and pooled size to previously produced pooled file from file 2. \n",
    "    *Simply adding two more columns to already exisintg columns\n",
    "    \n",
    "###### f. File5: geneiousanalysis2_pooled/EP.ipynb\n",
    "    1. This jupyter note file is for processing adding pooled group and pooled size to previously produced pooled file from file 3 which contains both individual and pooled information. \n",
    "    *Simply adding two more columns to already exisintg columns\n",
    "    \n",
    "    \n",
    "###### g. File6: weight_geneious_LAB/EP_fixed1.ipynb\n",
    "    1. Last step of producing weighted vaf and total sample files\n",
    "    *The header of final output is \"Site,G_annotation,Gene,Number of Samples,SNP,WVAF,Number of Mutants, Number of wildtypes\"\n",
    "    \n",
    "    \n",
    "For more specific description look into individual jupyter notebook scripts. It contains doc string to explain overall script and line by line comment for specific details.\n",
    "\n",
    "\n",
    "    \n",
    "    \n",
    "  \n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9aab58bc-bcb4-445f-a1c4-ef237d6857de",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "88a06cd5-5da2-4483-9a3a-abcbaf3d4997",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
