{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "e0180cc8-83fb-4f36-a63c-fc4176b031fa",
   "metadata": {},
   "outputs": [],
   "source": [
    "###Open raw output table file \"Annotations.csv\" from Geneious#####\n",
    "\n",
    "count220=0\n",
    "dict3={}\n",
    "with open (\"Annotations.csv\", \"r\") as r1:\n",
    "    with open(\"GuineaAnalysis.csv\" , \"w\") as w1:\n",
    "        w1.write(\"Sample,Pooled,SITE,GENE,SNP,COVERAGE,VAF,VF\\n\")\n",
    "        count=0\n",
    "        dict1={}\n",
    "        dict1[\"Ha\"]=\"Hamdalaye\"\n",
    "        dict1[\"Do\"]=\"Dorota\"\n",
    "        dict1[\"Ma\"]=\"Maferinyah\"\n",
    "        dict1[\"La\"]=\"Lay-Sarè\"\n",
    "        dict1[\"LS\"]=\"Lay-Sarè\"\n",
    "        dict2={}\n",
    "        for lines in r1:\n",
    "            \n",
    "            count+=1\n",
    "            \n",
    "            ######Check if there is Polymorphism in the line from the table#######\n",
    "            ######If there is polymorphism then that means there is variant######\n",
    "            ######According to Geneious as geneious report variants as polymorphisms####\n",
    "            \n",
    "            if \"Polymorphism\" in lines and \"Variants\" in lines:\n",
    "                ####sample name is first column right before the _ string#####\n",
    "                samplename=lines.split(\",\")[0].split(\"_\")[0]\n",
    "                ####Sample gene is the second coulmmn in the line####\n",
    "                samplegene=lines.split(\",\")[1]\n",
    "                if \"DHPS\" in samplegene:\n",
    "                    samplegene=\"DHPS\"\n",
    "                if \"mitochondrial\" in samplegene:\n",
    "                    samplegene=\"CYTB\"\n",
    "                ###Sample AA change is column 17 which has the amino acid changes####\n",
    "                sampleAAchange=lines.split(\",\")[16]\n",
    "                #sampleVAF=(lines.split(\"\\\"\")[0]).split(\",\")[-4]\n",
    "                #print(lines.split(\",\")[2])\n",
    "                #if sampleVAF!=\"\":\n",
    "                #    print(sampleVAF)\n",
    "                #print(sampleAAchange)\n",
    "                ####If there is Amino acid change reported combine it with the c variable which is codon position####\n",
    "                ####to create a SNP change report #####\n",
    "                if sampleAAchange!=\"\":\n",
    "                    a=sampleAAchange.split(\" \")[0]\n",
    "                    b=sampleAAchange.split(\" \")[2]\n",
    "                    #print(len(s))\n",
    "                #print(lines.split(\",\")[16])\n",
    "                    c=lines.split(\",\")[19]\n",
    "                    AAchangereport=(a+c+b)\n",
    "                    #print(AAchangereport)\n",
    "                    sampleVAF=(lines.split(\"M05039\")[0]).split(\",\")[-4]\n",
    "                    sampleVF=(lines.split(\"M05039\")[0]).split(\",\")[-2]\n",
    "                    #if samplename==\"19GNHa00xp003PfF1290_S85_L001_R_001\":\n",
    "                    #    print(lines.split(\"M05039\")[0].split(\",\"))\n",
    "                    sampleCoverage=lines.split(\",\")[27]\n",
    "                    #print(AAchangereport)\n",
    "                    if \"GN\" in samplename:\n",
    "                        #if \"A220S\" in lines:\n",
    "                        #    print(samplename)\n",
    "                        #if AAchangereport==\"A220S\" and \"Do\" in lines:\n",
    "                            #print(samplename,AAchangereport)\n",
    "                        #    if \"PfCRT\" in lines:\n",
    "                        #        count220+=1\n",
    "                        #    if \"PfCRT\" in lines:\n",
    "                                ###check in item is already in dictionary###\n",
    "                                #samplename,AAchangereport\n",
    "                                #if (samplename,AAchangereport) in dict3:\n",
    "                                #print(samplename,AAchangereport)\n",
    "                        #dict3[samplename,AAchangereport]=\"exist\"\n",
    "                        ####Use a dictionary to prevent overlaps######\n",
    "                        if samplename[9]==\"p\" and (samplename,AAchangereport) not in dict3:\n",
    "                            dict3[samplename,AAchangereport]=\"exist\"\n",
    "                            #if \"Do\" in lines and \"PfCRT\" in lines and AAchangereport==\"A220S\":\n",
    "                            #    print(samplename+\",\"+\"Pooled\"+\",\"+dict1[samplename[4:6]]+\",\"+samplegene+\",\"+AAchangereport)\n",
    "                            #if samplename==\"19GNHa00xp003PfF1290_S85_L001_R_001\":\n",
    "                            #    print(samplename+\",\"+\"Pooled\"+\",\"+dict1[samplename[4:6]]+\",\"+samplegene+\",\"+AAchangereport+\",\"+sampleCoverage+\",\"+sampleVAF+\",\"+sampleVF)\n",
    "                            w1.write(samplename+\",\"+\"Pooled\"+\",\"+dict1[samplename[4:6]]+\",\"+samplegene+\",\"+AAchangereport+\",\"+sampleCoverage+\",\"+sampleVAF+\",\"+sampleVF+\"\\n\")\n",
    "                        if samplename[9]!=\"p\" and (samplename,AAchangereport) not in dict3:\n",
    "                            dict3[samplename,AAchangereport]=\"exist\"\n",
    "                            w1.write(samplename+\",\"+\"Individual\"+\",\"+dict1[samplename[4:6]]+\",\"+samplegene+\",\"+AAchangereport+\",\"+sampleCoverage+\",\"+sampleVAF+\",\"+sampleVF+\"\\n\")\n",
    "                        #print(lines)\n",
    "                        #print(sampleVAF)\n",
    "with open (\"Annotations.csv\", \"r\") as r1:\n",
    "    with open(\"GuineaAnalysis.csv\" , \"a\") as w1:\n",
    "        for lines in r1:\n",
    "            if \"Coverage:\" in lines:\n",
    "                #print(lines)\n",
    "                samplename=lines.split(\",\")[0].split(\"_\")[0]\n",
    "                #print(samplename)\n",
    "                samplegene=lines.split(\",\")[1]\n",
    "                if \"DHPS\" in samplegene:\n",
    "                    samplegene=\"DHPS\"\n",
    "                if \"mitochondrial\" in samplegene:\n",
    "                    samplegene=\"CYTB\"\n",
    "                   \n",
    "                AAchangereport=lines.split(\",\")[-1].strip(\"\\n\")\n",
    "                sampleVAF=\"0%\"\n",
    "                sampleVF=\"0\"\n",
    "                #if samplename==\"19GNHa00xp003PfF1290_S85_L001_R_001\":\n",
    "                #    print(lines.split(\"M05039\")[0].split(\",\"))\n",
    "                sampleCoverage=lines.split(\",\")[-3]\n",
    "                #print(AAchangereport)\n",
    "                if \"GN\" in samplename:\n",
    "                    if samplename[9]==\"p\" and (samplename,AAchangereport) not in dict3:\n",
    "                        dict3[samplename,AAchangereport]=\"exist\"\n",
    "                        #if \"Do\" in lines and \"A220S\" in lines:\n",
    "                        #    print(lines)\n",
    "                        w1.write(samplename+\",\"+\"Pooled\"+\",\"+dict1[samplename[4:6]]+\",\"+samplegene+\",\"+AAchangereport+\",\"+sampleCoverage+\",\"+sampleVAF+\",\"+sampleVF+\"\\n\")\n",
    "                    if samplename[9]!=\"p\" and (samplename,AAchangereport) not in dict3:\n",
    "                        dict3[samplename,AAchangereport]=\"exist\"\n",
    "                        #print(samplename)\n",
    "                        w1.write(samplename+\",\"+\"Individual\"+\",\"+dict1[samplename[4:6]]+\",\"+samplegene+\",\"+AAchangereport+\",\"+sampleCoverage+\",\"+sampleVAF+\",\"+sampleVF+\"\\n\")\n",
    "                    #print(lines)\n",
    "                        #print(sampleVAF)\n",
    "#print(dict2)\n",
    "#print(count220)\n",
    "#print(len(dict3))\n",
    "##print(dict3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "42c87a95-20b3-49aa-b94d-3c1b17fd9d57",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b75b4d65-1f6e-4aae-a6af-723a04594ea1",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
